torch/csrc/jit/passes/inliner.cpp

```diff
@@ -12,23 +12,26 @@ using namespace ::c10::prim;
 void inlineCalls(Block* block) {
   for (auto it = block->nodes().begin(), end = block->nodes().end();
        it != end;) {
-    Node* cur = *it++;
+    Node* cur = *it;
     switch (cur->kind()) {
       case prim::CallFunction: {
         AT_ASSERT(cur->input(0)->node()->kind() == prim::Constant);
         auto function_constant = cur->input(0)->node();
         auto fun_type =
             function_constant->output()->type()->expect<FunctionType>();
         cur->removeInput(0);
+        // We need to decrerement `it` because inlineCallTo will destroy `cur`
+        // and potentially replace it with new nodes.
+        it--;
         inlineCallTo(cur, *fun_type->function()->graph());
-        if (!function_constant->hasUses()) {
```

Comment:
I removed this because it complicates the `it--` logic above. I think it's okay to litter these—we should be running DCE at some point anyway, I don't think needs to be a guarantee that function constants are cleaned up by the inliner. What do people think?

Reply:
> I removed this because it complicates the `it--` logic above. I think it's okay to litter these—we should be running DCE at some point anyway, I don't think needs to be a guarantee that function constants are cleaned up by the inliner. What do people think?

You could save these constants into some `to_remove` list and wipe them all off in the end. But I agree that we would want to run cleanup passes after inlining anyway, so up to you.

Reply:
@suo The historical reason for this logic is that I needed to make sure that after **Inliner's** inlining, graphs would be identical to the graphs after inlining in `compiler.cpp.` I couldn't run DCE since it would remove more than just those constants. Letting DCE handle this seems like the right thing to do.